### PR TITLE
feat: [CDS-42429]: customDeployment support in services 

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

Mocks: https://www.figma.com/proto/KAwaeKo8oh1qcrrTWt09y2/Deployment-Templates-3?node-id=2773%3A461669&scaling=min-zoom&page-id=2672%3A489450&starting-point-node-id=2773%3A461669&show-proto-sidebar=1

- Added service definition type to be Deployment Template Type
- Support for only Artifact and Variables added (v1) as per design

#### Screenshots

https://user-images.githubusercontent.com/100121280/190068656-239aa988-74b0-4bf9-8d1a-b204ac46572e.mov

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 144 Issues 
* Code issue 144 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11054)

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceDefinition/DeployServiceDefinition.tsx [hunk 1]
```diff
@@ -152,6 +164,40 @@ function DeployServiceDefinition(): React.ReactElement {
     }
   }
 
+  const addOrUpdateTemplate = async (): Promise<void> => {
```
Comment (vkbansal-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceDefinition/DeployServiceDefinition.tsx:167]: Changes also required for v2? cc @bhavya-sinha

Comment (bhavya-sinha) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceDefinition/DeployServiceDefinition.tsx:167]: This change is for service V2 creation, when user selects deployment type

Comment (bhavya-sinha) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceDefinition/DeployServiceDefinition.tsx:167]: @PratyushGar This component is rendered inside pipeline also, when user edits the service. (deployment type is disabled there though).
Please check if this is not opening drawer in pipeline modal view.

Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceDefinition/DeployServiceDefinition.tsx:167]: The behavior will be consistent on the stage addition popover as well, as we would require the drawer to be opened there as well. Thanks for the findings, Edit service modal changes will be part of the follow-up PR, and the same deployment template will be linked by default which the user selected on the stage edition popover, will ensure to check there as well.

https://user-images.githubusercontent.com/100121280/188795656-27603a5c-318d-4f9d-8461-0f14158c08bb.mov

**Review by samarth-sharma31 — APPROVED** [review]

**Review by yc147 — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**